### PR TITLE
std: add support for getting the time in UEFI

### PR DESCRIPTION
**Pull request by RossComputerGuy**
Adds some extra logic and things to make time work with UEFI. It may not be optimal but should do its job.

### CODE REVIEW

File: lib/std/time.zig [hunk 1]
```diff
@@ -114,6 +114,13 @@ pub fn nanoTimestamp() i128 {
         return ns;
     }
 
+    if (builtin.os.tag == .uefi) {
+        var value: std.os.uefi.Time = undefined;
+        const status = std.os.uefi.system_table.runtime_services.getTime(&value, null);
+        assert(status == .Success);
+        return value.toEpoch();
```
Comment (Vexu) [inline lib/std/time.zig:121]: `toEpoch` returns seconds but this `nanoTimestamp` should return nanoseconds.

**Review by Vexu — CHANGES_REQUESTED** [review]

File: lib/std/time.zig (hunk 1 above)
Comment (RossComputerGuy) [reply lib/std/time.zig]: Ok, that should be fixed now.

Comment (RossComputerGuy) [reply lib/std/time.zig]: Actually, I just realized UEFI time has nanoseconds so I'll change it to include that.

File: lib/std/os/uefi.zig [hunk 2]
```diff
@@ -131,6 +131,31 @@ pub const Time = extern struct {
...
+
+        while (year < (self.year - 1971)) : (year += 1) {
+            days += daysInYear(year + 1970, 12);
+        }
+
+        days += daysInYear(self.year, @as(u4, @intCast(self.month)) - 1) + self.day;
+        const hours = self.hour + (days * 24);
+        const minutes = self.minute + (hours * 60);
+        const seconds = self.second + (minutes * 60);
+        return self.nanosecond + (seconds * 1000);
```
Comment (truemedian) [inline lib/std/os/uefi.zig:157]: You should be using the std.time constants, there are more than `1000` nanoseconds in a second.

**Review by truemedian — CHANGES_REQUESTED** [review]

File: lib/std/os/uefi.zig (hunk 2 above)
Comment (RossComputerGuy) [reply lib/std/os/uefi.zig]: Ok, it should be right now.

**Pull request merged**